specs/corla.pvs

```diff
@@ -552,106 +998,364 @@ BEGIN
+%%% COUNTYWIDE CONTEST IN ANY COUNTY. THE SECRETARY OF STATE WILL
+%%% PUBLISH A COMPLETE LIST OF ALL AUDITED CONTESTS ON THE AUDIT
+%%% CENTER. THE SECRETARY OF STATE WILL CONSIDER THE FOLLOWING FACTORS
+%%% IN DETERMINING WHICH CONTESTS TO AUDIT:
+%%% (1) THE CLOSENESS OF THE REPORTED TABULATION OUTCOME OF THE CONTESTS;
+%%% (2) THE GEOGRAPHICAL SCOPE OF THE CONTESTS;
+%%% (3) ANY CAUSE FOR CONCERN REGARDING THE ACCURACY OF THE REPORTED
+%%% TABULATION OUTCOME OF THE CONTESTS;
+%%% (4) ANY BENEFITS THAT MAY RESULT FROM OPPORTUNISTICALLY AUDITING
+%%% CERTAIN CONTESTS; AND
+%%% (5) THE ABILITY OF THE COUNTY CLERKS TO COMPLETE THE AUDIT BEFORE
+%%% THE CANVASS DEADLINE.
+
+  audit_reason: TYPE =
+    { state_wide_contest, county_wide_contest, close_contest,
+      geographical_scope, concern_regarding_accuracy,
+      opportunitic_benefits, county_clerk_ability }
+
+  select_contest_for_comparison_audit:
+    [contest, audit_reason -> audited_contest]
+% @trace RDPR-6-Jul-2017 Section 25.2.2(I)
+% If there is at least one statewide contest, then at least one has to
+% be selected.
+  at_least_one_statewide_contest: AXIOM TBD
+% If there is at least one countywide contest, then at least one has
+% to be selected.
+  at_least_one_countywide_contest_per_county: AXIOM TBD
+
+  select_contest_for_opportunistic_audit:
+    [contest, audit_reason -> opportunistic_contest]
+
+%%% Ballots can be randomly selected for audit in two ways, either by:
+%%% \begin{enumeration}
+%%% \item permuting all ballots and auditing a prefix of ballots
+%%% (thereby auditing ballots "with no replacement")
+%%% \item randomly selecting ballots ballot-by-ballot (thereby
+%%% auditing ballots "with replacement" as the same ballot may be
+%%% audited multiple times).
+%%% \end{enumeration}
 
+% @todo kiniry Obviously this type has to be strengthened considerably
+% to guarantee permutation. I.e., both the set and the list are
+% finite and converting the list into a set results in a set equivalent
+% to the original set.
+  ballot_permutation:
+    [set[ballot] -> list[ballot]]
+% Note that sequences are infinite.
+  random_list_of_ballots:
+    [set[ballot] -> sequence[ballot]]
+
+% @trace RDPR-6-Jul-2017 Section 25.2.2(K)
+
+%%% \subsection{Random Selection of Ballots for Audit}
+
+%%% The Secretary of State will randomly select the individual ballots
+%%% to audit. The Secretary of State will use a pseudo-random number
+%%% generator with the seed established under subsection (H) of this
+%%% rule to identify individual ballots as reflected in the county
+%%% ballot manifests. The Secretary of State will notify each county
+%%% of, and publish on the audit center, the randomly selected ballots
+%%% that each county must audit no later than 11:59 p.m. MT on the
+%%% tenth day after Election Day.
+  public_ballot_to_audit:
+    [state_wide_dashboard, list[ballot] -> state_wide_dashboard]
+  
 END state_wide_dashboard
```

Comment:
We should add a `full_hand_count_contest` type, and a UI interaction which signals a human determination that an audited contest should go to a full hand count. Such contests, even though they have not reached their risk limit, should nevertheless no longer be audited via the tool. That's a human decision, made by either SoS or the clerks. @review CDOS Who should enter this determination into the audit tool?